### ISSUE TITLE
LUIS fails to initialize properties on FormFlow dialogs when the property type is a string.

### ISSUE THREAD

**Issue by jockorob** [Microsoft/BotBuilder#151]
Enums work great, I haven't tried ints, but strings don't work.  I believe the problem is in FormDialog.cs in the StartAsync() function.  

                    var matches = MatchAnalyzer.Coalesce(step.Match(context, _state, _formState, input), input);
                    if (MatchAnalyzer.IsFullMatch(input, matches, 0.5))
                    {
                        // TODO: In the case of clarification I could
                        // 1) Go through them while supporting only quit or back and reset
                        // 2) Drop them
                        // 3) Just pick one (found in form.StepState, but that is opaque here)
                        var result = await step.ProcessAsync(context, _state, _formState, input, matches);
                        feedback = result.Feedback;
                        prompt = result.Prompt;
                    }
                    else
                    {
                        _formState.SetPhase(StepPhase.Ready);
                    }

There are never matches when the property is a string like there are when the property is an enum.

**Comment by IanBoggs** [comment]
Just found this issue myself. 
Whilst this is great for enum properties, my evolving bot is definitely going to need strings

**Comment by chrimc62** [comment]
I believe this was fixed by the pull request I just accepted from @jockorob.  The issue was not strings, but that there was a mismatch between the LUIS model and the field name.

**Comment by IanBoggs** [comment]
Hi @chrimc62 ,

I do not believe that this was fixed in the pull request you mentioned. That pull request seems to fix an issue with the Luis model not matching the field name, which is not my case. 

Although that pull request does fix the mis-matched name, the problem still exists that string fields in a FormFlow dialog are not correctly matched. In the code quoted by @jockorob, if the field is a string it always executes the 'else' part of the 'if' statement.

I've tested this both with my own code and with the PizzaBot from the new pull. The Address field is still not populated when there is a Luis Entity called 'Address'

Can this issue be re-opened, or should I start a new issue ?
